src/crafting.cpp

```diff
@@ -1096,10 +1096,12 @@ bool item::handle_craft_failure( Character &crafter )
     // Loss is scaled by the success roll
     const double percent_progress_loss = rng_exponential( 0.25, 0.35 ) *
                                          ( 1.0 - std::min( success_roll, 1.0 ) );
```

Comment:
```suggestion
    const double percent_progress_loss = rng_exponential( 0.25, 0.35 ) * ( 1.0 - success_roll );
```
Since it isn't used if below 0.

Reply:
You are right mathematically, good catch. I am a bit wary because it loses the property of being a loss in that case, but maybe a comment up front is enough to clarify that. The std::min was actually added as a bugfix two years ago to prevent negative loss (or random progress advances during "failures"), but my solution fixes that as well as you noted. Just have to ensure that future commiters are warned up front about this.

Reply:
Done, plus added a new comment and edited an existing one for greater clarity.